pkg/reaper/nodereaper/nodereaper.go

```diff
@@ -551,6 +554,23 @@ func (ctx *ReaperContext) reapOldNodes(w ReaperAwsAuth) error {
 		if ctx.DryRun {
 			log.Warnf("dry run is on, '%v' will not be cordon/drained", instance.NodeName)
 		}
+
+		// TEST
+		err = obtainReapLock(w.DDB, instance.NodeName, instance.InstanceID, "master")
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				if aerr.Code() == dynamodb.ErrCodeConditionalCheckFailedException {
+					log.Infof("another master roll is in progress, skipping node %s (%s)", instance.NodeName, instance.InstanceID)
+				} else {
+					log.Infof("AWS error while attempting to obtain lock for %s (%s), skipping: %s", instance.NodeName, instance.InstanceID, aerr.Message())
+				}
+			} else {
+				log.Infof("unknown error while attempting to obtain lock for %s (%s), skipping: %s", instance.NodeName, instance.InstanceID, err.Error())
+			}
+
+			continue
+		}
```

Comment:
I wonder if we should check for a new master that is ready before releasing the lock instead of depending on the TTL of the ddb lock. That way the expiration is just a failure case, and we can have a bit quicker termination

Reply:
Hmm yeah we could refactor the `allNodesAreReady` helper func to take the node type 🤔